### PR TITLE
Fix: Adding pagination and correcting all user retrieval

### PR DESCRIPTION
**Pull request by AngeloChristian1**
#### What does this PR do?
It fixes issues that were occurring in data retrieval

#### Description of Task to be completed?

- Added pagination to all get All endpoints
- Removed password from being sent when users are being retrieved
- 

#### How should this be manually tested?
On swagger visited relevant pages for testing

### CODE REVIEW

File: src/controllers/adminController.ts [hunk 1]
```diff
@@ -3,22 +3,48 @@ import User from '../database/models/userModel'
...
+
+const {offset, limit, page} = paginationResults
+
+const totalCount: number = await getUsersCount()
+const totalPages = Math.ceil(totalCount / limit)
+
+    const users = await User.findAll({
+      attributes: { exclude: ['password'] },
+      offset,
+      limit
```
Comment (mpanoathos) [inline src/controllers/adminController.ts:35]: Nice

File: src/controllers/productController.ts [hunk 2]
```diff
@@ -144,42 +145,52 @@ static async listAllCollections(
...
+    //   Number.isNaN(limit) ||
+    //   page <= 0 ||
+    //   limit <= 0
+    // ) {
+    //   return res
+    //     .status(400)
+    //     .json({ message: 'Invalid pagination parameters' })
+    // }
+
+    // const offset = (page - 1) * limit
```
Comment (mpanoathos) [inline src/controllers/productController.ts]: Please remove commented codes since they are not performing anything

File: src/routes/userRoute.ts [hunk 3]
```diff
@@ -69,7 +69,7 @@ router.post(
 )
 router.get(
   '/',
-  isAuthenticated, checkPermission(UserRole.ADMIN),
+  // isAuthenticated, checkPermission(UserRole.ADMIN),
```
Comment (mpanoathos) [inline src/routes/userRoute.ts]: Remember to remove this comments, this action needs to be performed by admin

File: src/services/notificationServices.ts [hunk 4]
```diff
@@ -105,3 +106,13 @@ eventEmitter.on('order:updatedStatus', async (order: Order ) => {
...
+
+
+export const getNotificationCount = async (userId:number|UUID) => {
+
+  const users = await Notification.findAll({
+    where: { userId },
+    order: [['createdAt', 'DESC']],  
+  })
+  return users.length
+}
```
Comment (mpanoathos) [inline src/services/notificationServices.ts:118]: Nice

**Review by mpanoathos — CHANGES_REQUESTED** [review]
Great work! but there are some changes to make

**Review by mpanoathos — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**